Replicator/ReplicatorOptions.hh

```diff
@@ -298,32 +298,21 @@ namespace litecore { namespace repl {
             DebugAssert(!_isActive);
 
             size_t origCount = collectionOpts.size();
-            std::vector<C4CollectionSpec> specs;
-            specs.reserve(origCount);
-            for (auto& opt : collectionOpts) {
-                specs.emplace_back(collectionPathToSpec(opt.collectionPath));
-            }
-            for (size_t i = 0; i < activeCollections.size(); ++i) {
-                auto findIt = _collectionSpecToIndex.find(activeCollections[i]);
-                if (findIt == _collectionSpecToIndex.end()) {
-                    // Put the unfound spec to the end of collectionOpts
-                    size_t j = collectionOpts.size();
-                    collectionOpts.emplace_back(nullslice);
-                    // Assertion: collectionOpts[j].collectionPath == nullslice
-                    _collectionSpecToIndex[specs[i]] = j;
-                    std::swap(collectionOpts[i], collectionOpts[j]);
+            decltype(collectionOpts) reordered;
+            reordered.reserve(origCount);
+
+            for (size_t activeIndex = 0; activeIndex < activeCollections.size(); ++activeIndex) {
+                auto foundEntry = _collectionSpecToIndex.find(activeCollections[activeIndex]);
+                if (foundEntry == _collectionSpecToIndex.end()) {
+                    reordered.emplace_back(nullslice);
                 } else {
-                    DebugAssert(findIt->second >= i && findIt->second < origCount);
-                    if (findIt->second > i) {
-                        C4CollectionSpec spec;
-                        size_t           indx;
-                        std::tie(spec, indx) = *findIt;
-                        _collectionSpecToIndex[specs[i]] = indx;
-                        _collectionSpecToIndex[spec] = i;
-                        std::swap(collectionOpts[i], collectionOpts[indx]);
-                    }
+                    reordered.push_back(collectionOpts[foundEntry->second]);
                 }
+
+                _collectionSpecToIndex[activeCollections[activeIndex]] = activeIndex;
```

Comment:
When not matched, activeCollections[activeIndex] won't be in reordered and ultimately the updated collectionOpts, and, so, should be in the map.

Reply:
I didn't follow what you want to change.

Reply:
_collectionSpecToIndex maps spec to index in vector collectionOpts. When not matched, activeCollections[activeIndex] won't exist in collectionOpts. Besides, the map has the specs that were in original collectionOpts, but are dropped out of "reordered". So they should be removed from the map.
With brutal-force, simplest is to recreate the index map.

Reply:
So....don't include them in that case?  It's sort of splitting hairs because the map will never even be used since the entire replication is invalid.

Reply:
correct.